rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,217 @@
+        federationStorageProvider.setNewFederation(federation);
+        return federationStorageProvider;
+    }
+
+    private static FeePerKbSupport getFeePerKbSupport() {
+        FeePerKbSupport feePerKbSupport = mock(FeePerKbSupport.class);
+        Coin feePerKb = Coin.valueOf(1000L);
+        when(feePerKbSupport.getFeePerKb()).thenReturn(feePerKb);
+        return feePerKbSupport;
+    }
+
+    private static Block getRskExecutionBlock() {
+        long rskExecutionBlockNumber = 1000L;
+        long rskExecutionBlockTimestamp = 10L;
+        BlockHeader blockHeader = new BlockHeaderBuilder(mock(ActivationConfig.class))
+                .setNumber(rskExecutionBlockNumber)
+                .setTimestamp(rskExecutionBlockTimestamp)
+                .build();
+        return Block.createBlockFromHeader(blockHeader, true);
+    }
+
+    private BridgeSupport getBridgeSupport(BridgeStorageProvider bridgeStorageProvider, ActivationConfig.ForBlock activationsBeforeForks, FederationSupport federationSupport, FeePerKbSupport feePerKbSupport, Block rskExecutionBlock, BtcBlockStoreWithCache.Factory btcBlockStoreFactory, Repository repository, BtcLockSenderProvider btcLockSenderProvider) {
+        return bridgeSupportBuilder
+                .withBridgeConstants(bridgeConstants)
+                .withProvider(bridgeStorageProvider)
+                .withActivations(activationsBeforeForks)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withExecutionBlock(rskExecutionBlock)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withRepository(repository)
+                .withBtcLockSenderProvider(btcLockSenderProvider)
+                .build();
+    }
+
+
+    private FederationStorageProvider createFederationStorageProvider(Repository repository) {
+        StorageAccessor bridgeStorageAccessor = new BridgeStorageAccessorImpl(repository);
+        return new FederationStorageProviderImpl(bridgeStorageAccessor);
+    }
+
+    private BtcTransaction createPegInTransaction(Address federationAddress, Coin coin, BtcECKey pubKey) {
+        BtcTransaction btcTx = new BtcTransaction(btcParams);
+        btcTx.addInput(BitcoinTestUtils.createHash(0), 0, ScriptBuilder.createInputScript(null, pubKey));
+        btcTx.addOutput(new TransactionOutput(btcParams, btcTx, coin, federationAddress));
+
+        return btcTx;
+    }
+
+    private static PartialMerkleTree createValidPmtForTransactions(List<Sha256Hash> hashesToAdd, NetworkParameters networkParameters) {
+        byte[] relevantNodesBits = new byte[(int)Math.ceil(hashesToAdd.size() / 8.0)];
+        for (int i = 0; i < hashesToAdd.size(); i++) {
+            Utils.setBitLE(relevantNodesBits, i);
+        }
+
+        return PartialMerkleTree.buildFromLeaves(networkParameters, relevantNodesBits, hashesToAdd);
+    }
+
+    private static void recreateChainFromPmt(
```

Comment:
This type of methods would be better if they are in an utils class where we can reuse them later.

Reply:
I brought many of them from branches that are still not merged. I am expecting  them to be merged and then I can extract the methods to a Utils class